xwiki-platform-core/xwiki-platform-oldcore/src/test/java/com/xpn/xwiki/pdf/impl/PdfExportImplTest.java

```diff
@@ -63,6 +65,7 @@ public void applyCSSWhenExistingStyleDefinedUsingShorthandNotation() throws Exce
         this.oldcore.getMocker().registerMockComponent(Environment.class);
         this.oldcore.getMocker().registerMockComponent(VelocityManager.class);
         this.oldcore.getMocker().registerMockComponent(XSLFORenderer.class, "fop");
+        this.oldcore.getMocker().registerComponent(XMLReaderFactory.class, new DefaultXMLReaderFactory());
```

Comment:
this is not great since DefaultXMLReaderFactory is a component (thus we should not use `new`).

Reply:
Not sure how to register the component then.

Reply:
Since this is an integration tests (not a unit test), I think you want to use the real component implementation and thus not mock it. So the way to do this is to add a `@ComponentList()` annotation to the class:

```
@ComponentList({
  org.xwiki.xml.internal.DefaultXMLReaderFactory
})
```

See https://dev.xwiki.org/xwiki/bin/view/Community/Testing/JavaUnitTesting/#HComponentManager for details.

thanks!